src/scene/vcPOI.cpp

```diff
@@ -1271,52 +1272,74 @@ void vcPOI::AddFillPolygonToScene(vcState *pProgramState, vcRenderData *pRenderD
+    for (int64_t i = 0; i < m_line.numPoints; ++i)
+      centerPoint += (m_line.pPoints[i] / (double)m_line.numPoints);
+    
+    // Rotate
+    udDoubleQuat rotate = vcGIS_GetQuaternion(pProgramState->geozone, centerPoint);
+    udDoubleQuat rotateInverse = rotate;
+    rotateInverse.inverse();
 
-    pVerts[0] = { center, defaultNormal, defaultUV };
-    for (int i = 0; i < m_line.numPoints; ++i)
-      pVerts[i + 1] = { udFloat3::create(m_line.pPoints[i] - m_line.pPoints[0]), defaultNormal, defaultUV };
+    udDouble3 *pModifiedVerts = udAllocType(udDouble3, m_line.numPoints, udAF_Zero);
+    for (int64_t i = 0; i < m_line.numPoints; ++i)
+      pModifiedVerts[i] = m_line.pPoints[0] + rotateInverse.apply(m_line.pPoints[i] - m_line.pPoints[0]);
 
-    for (int i = 0; i < m_line.numPoints; ++i)
+    // Generate Triangles
+    vcCDT_ProcessOrignal(pModifiedVerts, m_line.numPoints, std::vector< std::pair<const udDouble3 *, size_t> >(), min, max, &trianglePointList);
+       
+    int numPoints = (int)trianglePointList.size();
+
+    vcP3N3UV2Vertex *pVerts = udAllocType(vcP3N3UV2Vertex, numPoints, udAF_Zero);
+    uint32_t *pIndices = udAllocType(uint32_t, numPoints, udAF_Zero);
+
+    for (int64_t i = 0; i < (int64_t)trianglePointList.size(); ++i)
+    {
+      udFloat3 pos = udFloat3::create((float)trianglePointList[i].x, (float)trianglePointList[i].y, 0);      
+      pVerts[i] = { pos, defaultNormal, defaultUV };
+      pIndices[i] = (uint32_t)i;
+    }
+    
+    // Un-flatten 2D Result
+    for (int64_t i = 0; i < (int64_t)trianglePointList.size(); ++i)
     {
-      int indicesIndex = i * 3;
-      pIndices[indicesIndex + 0] = 0; // center
-      pIndices[indicesIndex + 1] = i + 1;
-      pIndices[indicesIndex + 2] = i + 2;
+      double closestDist = FLT_MAX;
+      float closestZ = 0;
+      for (int64_t pointIndex = 0; pointIndex < m_line.numPoints; ++pointIndex)
+      {
+        udDouble2 pos = (pModifiedVerts[pointIndex] - pModifiedVerts[0]).toVector2();
+    
+        double dist = udMag2<double>(pos - trianglePointList[i]);
+        if (dist < closestDist)
+        {
+          closestDist = dist;
+          closestZ = (float)(pModifiedVerts[pointIndex].z - pModifiedVerts[0].z);
+        }
+      }
+    
+      pVerts[i].position.z = closestZ;
     }
 
-    // Wrap last triangle around to first vertex
-    pIndices[m_line.numPoints * 3 - 1] = 1;
+    // Un-Rotate
+    udFloatQuat rotatef = udFloatQuat::create(rotate);
+    for (int64_t i = 0; i < (int64_t)trianglePointList.size(); ++i)
+      pVerts[i].position = rotatef.apply(pVerts[i].position);
 
     vcPolygonModel_Destroy(&m_pPolyModel);
-    vcPolygonModel_CreateFromRawVertexData(&m_pPolyModel, pVerts, numVerts, vcP3N3UV2VertexLayout, (int)(udLengthOf(vcP3N3UV2VertexLayout)), pIndices, numIndices);
+    vcPolygonModel_CreateFromRawVertexData(&m_pPolyModel, pVerts, numPoints, vcP3N3UV2VertexLayout, (int)(udLengthOf(vcP3N3UV2VertexLayout)), pIndices, numPoints);
```

Comment:
While this will work - having a 1:1 index to vertex ratio is just wasted memory (since no vertices are being reused).

Remove `pIndices` from this code completely.